scripts/build/builders/realtek.py

```
@@ -0,0 +1,171 @@
+
+
+class RealtekBuilder(Builder):
+
+    def __init__(self,
+                 root,
+                 runner,
+                 board: RealtekBoard = RealtekBoard.RTL8777G,
+                 app: RealtekApp = RealtekApp.LIGHT,
+                 enable_cli: bool = False,
+                 enable_rpc: bool = False,
+                 enable_shell: bool = False):
+        super(RealtekBuilder, self).__init__(root, runner)
+        self.board = board
+        self.app = app
+        self.enable_cli = enable_cli
+        self.enable_rpc = enable_rpc
+        self.enable_shell = enable_shell
+
+        os.environ['OT_SRCDIR'] = os.path.join(os.getcwd(), 'third_party/openthread/ot-realtek')
+        os.environ['REALTEK_SDK_PATH'] = os.path.join(os.environ['OT_SRCDIR'], 'third_party/Realtek/rtl87x2g_sdk')
+        os.environ['BUILD_BANK'] = 'bank0'
+        os.environ['OT_CMAKE_NINJA_TARGET'] = self.app.TargetName
+
+    def CmakeBuildFlags(self):
+        flags = []
+
+        flags.append("-DCMAKE_BUILD_TYPE=Release")
+        flags.append("-DCMAKE_TOOLCHAIN_FILE=src/bee4/arm-none-eabi.cmake")
+        flags.append("-DBUILD_TYPE=sdk")
+        flags.append(f"-DBUILD_TARGET={self.board.BoardName}")
+        flags.append(f"-DBUILD_BOARD_TARGET={self.board.BoardName}")
+        flags.append(f"-DOT_CMAKE_NINJA_TARGET={self.app.TargetName}")
+
+        if self.enable_cli:
+            flags.append("-DENABLE_CLI=ON")
+        else:
+            flags.append("-DENABLE_CLI=OFF")
+
+        if self.enable_rpc:
+            flags.append("-DENABLE_PW_RPC=ON")
+        else:
+            flags.append("-DENABLE_PW_RPC=OFF")
+
+        if self.enable_shell:
+            flags.append("-DENABLE_SHELL=ON")
+        else:
+            flags.append("-DENABLE_SHELL=OFF")
+
+        build_flags = " ".join(flags)
+
+        return build_flags
+
+    def generate(self):
+        os.environ['OUT_FOLDER'] = self.output_dir
+
+        cmd = 'cd {}/openthread \n'.format(os.environ['OT_SRCDIR'])
```

should we error out if OT_SRCDIR is not defined?